### DISCUSSION TITLE
Convert Indentation

### DISCUSSION THREAD

**Discussion by samualtnorman** [neovim/neovim#28722] [Q&A] [answered]
I'm trying to switch from VS Code to Neovim. VS Code has a command called "Convert Indentation to Tabs" and a command called "Convert Indentation to Spaces". I'm wondering if Neovim has equivalent commands built in or a plugin that provides equivalent commands.

**Comment by clason** [comment]
`:h retab`, controlled by your respective options.

**Reply by samualtnorman** [reply]
I can't seem to get this to convert spaces to tab characters

**Reply by clason** [reply]
If you want bidirectional conversion, that is not something Neovim provides. I'm sure there's a plugin for it. (It's also fairly simple to script yourself.)

**Reply by simonmandlik** [reply]
I use

```
:set expandtab
:%retab -indentonly
```

to convert leading tabs to spaces, and

```
:set noexpandtab
:%retab! -indentonly
```

to go back.

**Comment by samualtnorman** [comment, marked as answer]
I figured it out.

For converting spaces to tabs:
1.  Make sure you `:set shiftwidth=<number>` and `:set tabstop=<number>` to what the file is currently using.
2. `:set noexpandtab`
3. Select the whole file in visual mode
4. `>`
5. Select the whole file in visual mode
6. `<`

**Reply by clason** [reply]
If that's what you're after, `gg=G` is shorter.

**Reply by samualtnorman** [reply]
This works, thank you. How does it work?

**Reply by clason** [reply]
`:h =` (and `gg` goes to the top of the file, while `G` is a motion to the end of the file)
